docs/registry/README.md

```diff
@@ -0,0 +1,194 @@
+  name: github-source-sample
+  namespace: default
+spec:
+  eventTypes:
+    - push
+    - pull_request
+  ownerAndRepository: my-other-user/my-other-repo
+  accessToken:
+    secretKeyRef:
+      name: github-secret
+      key: accessToken
+  secretToken:
+    secretKeyRef:
+      name: github-secret
+      key: secretToken
+  sink:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: Broker
+    name: default
+
+```
+ 
+By applying this file, two EventTypes will be registered, with types `push` and `pull_request`, 
+source `my-other-user/my-other-repo`, for the `default` Broker in the `default` namespace.
+
+**2. Manual User Registration**
+
+A user manually `kubectl applies` an EventType CR.
+
+Example: 
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repofork
+  namespace: default
+spec:
+  type: repo:fork
+  source: my-other-user/my-other-repo
+  broker: dev
+``` 
+
+This would register the EventType named `repofork` with type `repo:fork`, source `my-other-user/my-other-repo` 
+in the `dev` Broker of the `default` namespace.
+
+
+**3. Broker Auto Registration Policy**
+
+Upon arrival of a non-registered EventType to a Broker ingress, and in case the Broker 
+ingress policy allows auto-registration of EventTypes, the Broker will create the EventType.
+
+Example: 
+
+Set up a Broker with `autoAdd` enabled.
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: auto-add-demo
+spec:
+  ingressPolicy:
+    autoAdd: true
+```
+
+Now if someone emits a non-registered event (e.g., `dev.knative.foo.bar`)
+into the Broker `auto-add-demo`, an EventType will be created upon the event arrival. 
+Note that the Broker's address is well-known, it will always be
+`<name>-broker.<namespace>.svc.<ending>`. In this case case, it is
+`auto-add-demo-broker.default.svc.cluster.local`.
+
+We can send the event manually. While SSHed into a `Pod` with the Istio sidecar, run:
+
+```shell
+curl -v "http://auto-add-demo-broker.default.svc.cluster.local/" \
+  -X POST \
+  -H "X-B3-Flags: 1" \
+  -H "CE-CloudEventsVersion: 0.1" \
+  -H "CE-EventType: dev.knative.foo.bar" \
+  -H "CE-EventTime: 2018-04-05T03:56:24Z" \
+  -H "CE-EventID: 45a8b444-3213-4758-be3f-540bf93f85ff" \
+  -H "CE-Source: dev.knative.example" \
+  -H 'Content-Type: application/json' \
+  -d '{ "much": "wow" }'
+```
+ 
+The Broker `auto-add-demo` will then create the EventType with type `dev.knative.foo.bar` in the Registry.
+
+## Discoverability Use Case
+
+By adding the spec.* fields of EventType as custom columns in the CRD we can fulfill the discoverability 
+use case: 
+
+*As an Event Consumer I want to be able to discover the different event types that I can consume 
+from the different Brokers, without resorting to any OOB mechanism.*
+
+First, Event Consumers will list the EventTypes registered in the system
+
+`$ kubectl get eventtypes -n default`
+
+
+NAME | TYPE | SOURCE | SCHEMA | BROKER | READY | REASON
```

Comment:
This might be clearer as a preformatted block of what kubectl will actually emit rather than a markdown table. Then the reader knows that they should expect this as output to the `kubectl get eventtypes` command.

Reply:
Makes sense! Done, thanks!